### ISSUE TITLE
hdfs source connector

### ISSUE THREAD

**Issue by zqhxuyuan** [confluentinc/kafka-connect-hdfs#146]
this project is aim to write to hdfs, is there a source connector that read from hdfs (then write to kafka)?

**Comment by cotedm** [comment]
@zqhxuyuan not currently based on what we know about [here](http://connectors.confluent.io). That said, maybe you can flesh out your idea a bit at least in terms of the use case? Typically HDFS is more of a "final resting place" for data in my experience because of it's huge capacity and high latency for extraction. However, if you have a use case it would be interesting to hear about.

**Comment by skonto** [comment]
@cotedm IMHO there are frameworks like Spark which support loading data from such sources. There are practical cases like when experimenting with data that would be great not to have to use another tool for ETL and stay within the kafka ecosystem.

**Comment by ewencp** [comment]
A source connector would be a bit unusual, but we've heard of use cases for HDFS and S3 source connectors to be used as a backup mechanism -- it might be an infrequently used connector, but could be used for reloading data in the event you somehow lost your data in Kafka, i.e. a disaster recovery solution. We don't have immediate plans for writing these connectors, but we are aware of a few requests for them. Just as we have both source and sink connectors in the JDBC repository, we'd be happy to house both in this repository as well.

**Comment by skonto** [comment]
Ok do you think it would be big effort? I have looked into the source code in the past but it was a bit long ago.

**Comment by ewencp** [comment]
I doubt it would be very complicated. Unlike the sink where the logic is relatively complicated for getting exactly once delivery, the source connector would look much like other filesystem connectors. The connector thread would probably be responsible for monitoring HDFS for files as they appear and then assigning them to tasks for processing (or define a way for them to split up the work amongst themselves automatically).

Then the tasks would simply process the files in order, reading each record from each file, generating a record for each, and using a combination of filename + offset in file as the records offset. A good basic reference for a related connector would be this [spooldir file connector](https://github.com/jcustenborder/kafka-connect-spooldir).

**Comment by skonto** [comment]
Ok sounds good. Thnx @ewencp

**Comment by srijiths** [comment]
I have created HDFS, S3, FTP like non-real time data sources. In my case , the use case is i want to join stream data with reference/master data which will be in any of the above non-real time connectors. I think this would be common usecase. Let me see if i can open source it.

**Comment by ewencp** [comment]
@skonto Ah, I was not aware they had added and HDFS support for that connector! I haven't tested the combination of that source and this sink, but it sounds like it could definitely work!

**Comment by pranayreddy105** [comment]
Hi Everyone,

I have some files in hdfs one location and I want to move from it to other hdfs two location in the cluster.
For this we wrote a shell script which keeps looking  one location, if it finds any new file added then it copies to the hdfs two location.

But here, I want to use kafka in between , is it possible
source is hdfs and destination is hdfs
I found there is no source connector for hdfs.. so how can I achieve this?
